library/src/main/java/com/novoda/downloadmanager/lib/BatchRetrievalException.java

```diff
@@ -0,0 +1,15 @@
+package com.novoda.downloadmanager.lib;
+
+final class BatchRetrievalException extends RuntimeException {
+    static BatchRetrievalException failedQueryForBatches() {
+        return new BatchRetrievalException("Failed to query for batches");
+    }
+
+    static BatchRetrievalException failedQueryForBatch(long batchId) {
```

Comment:
I think its a bit strange to have static factory methods on exceptions? Most people don't expect this with exceptions what about just using the constructor?

```java
public BatchRetrievalException() {
        super("Failed to query for batches");
    }

public BatchRetrievalException(long batchId) {
        super("Failed to query for batch with batchId = " + batchId);
    }
```